lib/spack/spack/solver/asp.py

```diff
@@ -1060,6 +1060,7 @@ def external_packages(self):
             for id, spec in enumerate(external_specs):
                 self.gen.newline()
                 spec_id = fn.external_spec(pkg_name, id)
+                self.possible_versions[spec.name].add(spec.version)
```

Comment:
This right here is an incredible testament to the rapidly appreciating future value of the clingo solver (imho). A one-line fix? This is *awesome!*

Reply:
The ASP syntax is indeed very compact. To give an example, we had for a while the issue of avoiding cycles in the spec graph. With `clingo` we can prevent such a graph from being constructed in just 3 lines:
```
path(Parent, Child) :- depends_on(Parent, Child).
path(Parent, Descendant) :- path(Parent, A), depends_on(A, Descendant).
:- path(A, B), path(B, A).
```